src/scripting/effects.cpp

```diff
@@ -15,14 +15,44 @@ inline uint32_t apply_subeffects(EFFECT_PARAMTERS) {
 
 	uint32_t i = 0;
 	while(sub_units_start < tval + source_size) {
-		i += internal_execute_effect(sub_units_start, ws, primary_slot, this_slot, from_slot, r_hi, r_lo + i);
+		i += internal_execute_effect(sub_units_start, ws, primary_slot, this_slot, from_slot, r_hi, r_lo + i, els);
 		sub_units_start += 1 + get_generic_effect_payload_size(sub_units_start);
 	}
 	return i;
 }
 
 uint32_t es_generic_scope(EFFECT_PARAMTERS) {
-	return apply_subeffects(tval, ws, primary_slot, this_slot, from_slot, r_hi, r_lo);
+	return apply_subeffects(tval, ws, primary_slot, this_slot, from_slot, r_hi, r_lo, els);
+}
+
+uint32_t es_if_scope(EFFECT_PARAMTERS) {
+	uint32_t ret = 0;
+	els = false;
+	if((tval[0] & effect::scope_has_limit) != 0) {
+		auto limit = trigger::payload(tval[2]).tr_id;
+		if(trigger::evaluate(ws, limit, primary_slot, this_slot, from_slot)) {
+			ret = apply_subeffects(tval, ws, primary_slot, this_slot, from_slot, r_hi, r_lo, els);
+			els = true;
+		}
+	} else {
+		ret = apply_subeffects(tval, ws, primary_slot, this_slot, from_slot, r_hi, r_lo, els);
+		els = true;
+	}
+	return ret;
+}
+uint32_t es_else_if_scope(EFFECT_PARAMTERS) {
+	uint32_t ret = 0;
+	if((tval[0] & effect::scope_has_limit) != 0) {
+		auto limit = trigger::payload(tval[2]).tr_id;
+		if(!els && trigger::evaluate(ws, limit, primary_slot, this_slot, from_slot)) {
+			ret = apply_subeffects(tval, ws, primary_slot, this_slot, from_slot, r_hi, r_lo, els);
+			els = true;
+		}
+	} else {
```

Comment:
needs to test for !els here, otherwise else will always run if there is no limit in it